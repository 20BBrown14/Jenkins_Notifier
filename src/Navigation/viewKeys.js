export const VK_ADD_NEW_JOB = 'add_new_job';
export const VK_JOB_STATUS = 'job_status';
