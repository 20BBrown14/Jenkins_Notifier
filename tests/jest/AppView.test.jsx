import React from 'react';
import View from '../../src/AppView';
import { VK_REPOS, VK_ADD_NEW_REPO } from '../../src/Navigation/viewKeys';

describe('AppView', () => {
  it('should render a default view', () => {
    const testView = (shallow(
      <View
        viewKey="someKey"
        repos={{}}
      />,
    ));
    expect(testView).toMatchSnapshot();
  });
  describe('When view key is VK_REPOS', () => {
    it('should render a view with a status component', () => {
      const testView = (shallow(
        <View
          viewKey={VK_REPOS}
          repos={{}}
        />,
      ));
      expect(testView).toMatchSnapshot();
    });
  });
  describe('When view key is VK_ADD_NEW_REPO', () => {
    it('should render a vie with an AddRepo component', () => {
      const testView = (shallow(
        <View
          viewKey={VK_ADD_NEW_REPO}
          repos={{}}
        />,
      ));
      expect(testView).toMatchSnapshot();
    });
  });
});
