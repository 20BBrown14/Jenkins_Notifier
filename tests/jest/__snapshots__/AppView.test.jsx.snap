// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView When view key is VK_ADD_NEW_REPO should render a vie with an AddRepo component 1`] = `
<div>
  <Connect(AddRepoContainer) />
</div>
`;

exports[`AppView When view key is VK_JOB_STATUS should render a view with a status component 1`] = `
<div>
  <Connect(StatusContainer)
    repos={Object {}}
  />
</div>
`;

exports[`AppView should render a default view 1`] = `<div />`;
